src/box/replication.cc

```diff
@@ -1258,6 +1297,10 @@ void
 replicaset_sync(void)
 {
 	int quorum = replicaset_sync_quorum();
+	double deadline;
+
+	if (replication_is_shutdown)
+		goto shutdown;
```

Comment:
I don't think this is necessary. AFAIU if replication is shutting down, the the loop condition below would make it break immediately anyway. For the same reason, I don't think we need to set `replication_fiber` in this function.

Reply:
Dropped changes in this function. Looks like this function will work as is because on stop applier is accounted for as sync. Also this function is not called before `replicaset_connect` which will fail after shutdown.